service/src/main/java/org/mskcc/cmo/metadb/service/SampleService.java

```diff
@@ -23,4 +23,6 @@
     List<SampleMetadata> getSampleMetadataListByCmoPatientId(String cmoPatientId) throws Exception;
 
     List<MetaDbSample> getAllMetadbSamplesByRequestId(String requestId) throws Exception;
+    
+    List<SampleMetadata> retrieveSampleMetadataHistory(String igoSampleId) throws Exception;
```

Comment:
methods names should be consistent with one another so this should be renamed  to something like `getSampleMetadataHistoryByIgoId`